code/modules/spells/spell_code.dm

```diff
@@ -99,9 +99,9 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 	return
 
 /spell/proc/is_valid_target(var/target, mob/user)
-	if(!(spell_flags & INCLUDEUSER) && target == usr)
+	if(!(spell_flags & INCLUDEUSER) && target == user)
 		return 0
-	if(get_dist(usr, target) > range) //Shouldn't be necessary but a good check in case of overrides
+	if(get_dist(user, target) > range && !(range == SELFCAST && target == user) && !(range == GLOBALCAST)) //Shouldn't be necessary but a good check in case of overrides
```

Comment:
That is certainly one way to do it, but I believe line 106 might be a problem when used in a general sense since it doesn't respect compatible_mobs